test/powershell/engine/Cdxml/Cdxml.Tests.ps1

```diff
@@ -0,0 +1,268 @@
+            $result = Get-Module CimTest
+            $result.ModuleBase | should be ${script:ModuleDir}
+        }
+
+        It "The CimTest module should have the proper cmdlets" @ItSkipOrPending {
+            $result = Get-Command -Module CimTest
+            $result.Count | Should Be 4
+            ($result.Name | sort-object ) -join "," | Should Be "Get-CimTest,New-CimTest,Remove-CimTest,Set-CimTest"
+        }
+    }
+
+    Context "Get-CimTest cmdlet" {
+        It "The Get-CimTest cmdlet should return 4 objects" @ItSkipOrPending {
+            $result = Get-CimTest
+            $result.Count | should be 4
+            ($result.id |sort-object) -join ","  | should be "1,2,3,4"
+        }
+
+        It "The Get-CimTest cmdlet should retrieve an object via id" @ItSkipOrPending {
+            $result = Get-CimTest -id 1
+            @($result).Count | should be 1
+            $result.field1 | Should be "instance 1"
+        }
+
+        It "The Get-CimTest cmdlet should retrieve an object by piped id" @ItSkipOrPending {
+            $result = 1,2,4 | foreach-object { [pscustomobject]@{ id = $_ } } | Get-CimTest
+            @($result).Count | should be 3
+            ( $result.id | sort-object ) -join "," | Should be "1,2,4"
+        }
+
+        It "The Get-CimTest cmdlet should return the proper error if the instance does not exist" @ItSkipOrPending {
+            { Get-CimTest -ea stop -id "ThisIdDoesNotExist" } | ShouldBeErrorId "CmdletizationQuery_NotFound_Id,Get-CimTest"
+        }
+
+        It "The Get-CimTest cmdlet should work as a job" @ItSkipOrPending {
+            try {
+                $job = Get-CimTest -AsJob
+                $result = $null
+                $i = 0
+                # wait up to 10 seconds, then the test will fail
+                # we need to wait long enough, but not too long
+                # the time can be adjusted
+                do {
+                    if ( $job.State -eq "Completed" ) 
+                    { 
+                        $result = $job | Receive-Job
+                        break 
+                    }
+                    start-sleep 1
+                } while ( $i++ -lt 10 )
+                $result.Count | should be 4
+                ( $result.id | sort-object ) -join "," | Should be "1,2,3,4"
+            }
+            finally {
+                if ( $job ) {
+                    $job | Remove-Job -force
+                }
+            }
+        }
+
+        It "Should be possible to invoke a method on an object returned by Get-CimTest" @ItSkipOrPending {
+            $result = Get-CimTest | Select-Object -first 1
+            $result.GetCimSessionInstanceId() | Should BeOfType [guid]
+        }
+    }
+
+    Context "Remove-CimTest cmdlet" {
+        BeforeEach {
+            Get-CimTest | Remove-CimTest
+            1..4 | %{ New-CimInstance -namespace root/default -class PSCore_Test1 -property @{  

```

Comment:
This needs to be addressed as we need to the PSScriptAnalyzer warnings and errors (#3791).